### ISSUE TITLE
set_rate_BC accepts values as an argument

### ISSUE THREAD

**Issue by jgostick** [PMEAL/OpenPNM#1518]
It might be more logical if this accepted ``rates`` , so:

``set_value_BC`` accepts ``values``

``set_rate_BC`` accepts ``rates``

Obviously this is very minor.  This could be done in a backward compatible way by accepted both arguments.

**Comment by jgostick** [comment]
```
def set_rate_BC(self, pores, rates, **kwargs):
    if 'values' in kwargs.keys():
        rates = kwargs.pop("values")
   ....proceed rest of func using 'rates'
```

**Comment by jgostick** [comment]
This can fix the present issues as well as issue #1340:
1.  Remove `value` argument
2.  Add a `rates` argument that only accepts arrays
3.  Add a `total_rate` argument that only accepts a scalar.

**Comment by jgostick** [comment]
While we're at it, fix the doc string to remove erroneous discussion of Neumann bc
